addons/crm/models/crm_team.py

```diff
@@ -113,12 +146,329 @@ def _alias_get_creation_values(self):
+
+        :param int work_days: see ``CrmTeam.action_assign_leads()``;
+
+        :return teams_data, members_data: structure-based result of assignment
+          process. For more details about data see ``CrmTeam._assign_leads()``
+          and ``CrmTeamMember._assign_and_convert_leads``;
+        """
+        if not self.env.user.has_group('sales_team.group_sale_manager') and not self.env.user.has_group('base.group_system'):
+            raise exceptions.UserError(_('Lead/Opportunities automatic assignment is limited to managers or administrators'))
+
+        _logger.info('### START Lead Assignment (%d teams, %d sales persons, %d work_days)' % (len(self), len(self.crm_team_member_ids), work_days))
+        teams_data = self._allocate_leads()
+        _logger.info('### Team repartition done. Starting salesmen assignment.')
+        members_data = self.crm_team_member_ids._assign_and_convert_leads(work_days=work_days)
+        for member_id, member_info in members_data.items():
+            _logger.info('-> member %s: assigned %s' % (member_id, member_info["assign"]))
+        _logger.info('### END Lead Assignment')
+        return teams_data, members_data
+
+    def _action_assign_leads_log(self, teams_data, members_data, html=False):
+        """ Tool method to prepare notification about assignment process result.
+
+        :param teams_data: see ``CrmTeam._allocate_leads()``;
+        :param members_data: see ``CrmTeamMember._assign_and_convert_leads()``;
+        :param html: if True target formatted content in html (suitable for a
+          log on a document); else target plaintext content (suitable for a
+          small notification).
+
+        :return string: formatted log
+        """
+        # extract some statistics
+        assigned = sum(len(teams_data[team]['assign']) + len(teams_data[team]['merged']) for team in self)
+        duplicates = sum(len(teams_data[team]['duplicates']) for team in self)
+        members = len(members_data.keys())
+        members_assigned = sum(len(member_data['assign']) for member_data in members_data.values())
+
+        # format user notification
+        message_parts = []
+        # 1- duplicates removal
+        if duplicates:
+            message_parts.append(_("%(duplicates)s duplicates leads have been merged.",
+                                   duplicates=duplicates))
+
+        # 2- nothing assigned at all
+        if not assigned and not members_assigned:
+            if len(self) == 1:
+                if not self.assignment_max:
+                    message_parts.append(
+                        _("No allocated leads to %(team_name)s team because it has no capacity. Add capacity to its salespersons.",
+                          team_name=self.name))
+                else:
+                    message_parts.append(
+                        _("No allocated leads to %(team_name)s team and its salespersons because no unassigned lead matches its domain.",
```

Comment:
Is that the only possible reason?
Maybe they were already at max capacity?

Reply:
Arf, could be true. I will check it to see if I can spot it.